### PR TITLE
EVG-5591: Add REST route for getting commit queue contents

### PR DESCRIPTION
**Pull request by ybrill**
Also preemptively changing variable names: CommitQ->CommitQueue

### CODE REVIEW

File: rest/model/commit_queue.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+type APICommitQueue struct {
+	ProjectID APIString   `json:"queue_id"`
+	Queue     []APIString `json:"queue"`
+}
+
+func (cq *APICommitQueue) BuildFromService(h interface{}) error {
+	cqService, ok := h.(commitqueue.CommitQueue)
+	if !ok {
+		return errors.New("incorrect type when converting commit queue")
```
Comment (bsamek) [inline rest/model/commit_queue.go]: return errors.Errorf("incorrect type `%T` when converting commit queue")

File: rest/route/commit_queue.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+func (cq commitQueueGetHandler) Factory() gimlet.RouteHandler {
+	return &commitQueueGetHandler{
+		sc: cq.sc,
+	}
+}
+
+func (cq commitQueueGetHandler) Parse(ctx context.Context, r *http.Request) error {
+	cq.project = gimlet.GetVars(r)["project_id"]
+	return nil
```
Comment (tychoish) [inline rest/route/commit_queue.go]: Should we return a 404 if the project doesn't exist?

File: rest/data/interface.go [hunk 3]
```diff
@@ -229,4 +230,5 @@ type Connector interface {
 	// Commit queue methods
 	GithubPREnqueueItem(string, string, int) error
 	EnqueueItem(string, string, string, string) error
+	FindCommitQueueByID(string) (*commitqueue.CommitQueue, error)
```
Comment (tychoish) [inline rest/data/interface.go]: I know we've been wildly inconsistent with this on the past but I've started to think that the connector should really not have interface dependencies on the database models and just return rest models. 

Would it be too much of a bother to change that here?

File: rest/model/commit_queue.go (hunk 1 above)
Comment (ybrill) [reply rest/model/commit_queue.go]: done.

File: rest/route/commit_queue.go (hunk 2 above)
Comment (ybrill) [reply rest/route/commit_queue.go]: done.

File: rest/data/interface.go (hunk 3 above)
Comment (ybrill) [reply rest/data/interface.go]: done.

File: rest/route/commit_queue.go [hunk 4]
```diff
@@ -36,12 +36,14 @@ func (cq commitQueueGetHandler) Parse(ctx context.Context, r *http.Request) erro
...
-		return gimlet.MakeJSONErrorResponder(errors.Wrap(err, "can't read commit queue into API model"))
+	if commitQueue == nil {
+		return gimlet.MakeJSONErrorResponder(gimlet.ErrorResponse{
+			StatusCode: http.StatusNotFound,
+			Message:    fmt.Sprintf("no matching project for ID %s", cq.project),
+		})
 	}
 
-	return gimlet.NewJSONResponse(apiCommitQueue)
+	return gimlet.NewJSONResponse(*commitQueue)
```
Comment (tychoish) [inline rest/route/commit_queue.go]: I don't think you need to de-reference this pointer...

**Pull request merged**